src/matrix.rs

```diff
@@ -106,10 +127,80 @@ impl<T:Add<T, Output=T> + Copy + Clone + ApproxEq<T> + Mul<T, Output=T> + One +
 
         return self.mul(&matrix);
     }
+
+    pub fn create_translation(x: T, y: T, z: T) -> Matrix4<T> {
+        let (_0, _1): (T, T) = (Zero::zero(), One::one());
+        Matrix4(_1.clone(), _0.clone(), _0.clone(), _0.clone(),
+                _0.clone(), _1.clone(), _0.clone(), _0.clone(),
+                _0.clone(), _0.clone(), _1.clone(), _0.clone(),
+                         x,          y,          z, _1.clone())
+    }
+
+    pub fn create_scale(x: T, y: T, z: T) -> Matrix4<T> {
+        let (_0, _1): (T, T) = (Zero::zero(), One::one());
+        Matrix4(         x, _0.clone(), _0.clone(), _0.clone(),
+                _0.clone(),          y, _0.clone(), _0.clone(),
+                _0.clone(), _0.clone(),          z, _0.clone(),
+                _0.clone(), _0.clone(), _0.clone(), _1.clone())
+    }
+
+    pub fn create_rotation(x: T, y: T, z: T, theta: T) -> Matrix4<T> {
```

Comment:
I guess x, y, and z are supposed to be normalized here? Might want to document that.